core/src/main/java/io/aiven/klaw/config/WebConfig.java

```diff
@@ -29,4 +32,12 @@ public void addResourceHandlers(ResourceHandlerRegistry registry) {
         .addResourceLocations("classpath:/static/assets/images/")
         .setCachePeriod(604800); // 1 week in seconds
   }
+
+  @Autowired private PermissionInterceptor permissionInterceptor;
+
+  @Override
+  public void addInterceptors(InterceptorRegistry registry) {
+    // Custom interceptor, add intercept path and exclude intercept path
+    registry.addInterceptor(permissionInterceptor).addPathPatterns("/**");
+  }
```

Comment:
Can we use AOP like below instead of interceptor ? 

package io.aiven.klaw.validation;

import io.aiven.klaw.error.PermissionConstraintException;
import io.aiven.klaw.model.enums.PermissionType;
import io.aiven.klaw.service.CommonUtilsService;
import org.aspectj.lang.annotation.Aspect;
import org.aspectj.lang.annotation.Before;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

import static io.aiven.klaw.helpers.UtilMethods.getPrincipal;

```
@Aspect
@Component
public class PermissionValidationAspect {
    @Autowired
    private CommonUtilsService commonUtilsService;

    @Value("${klaw.api.permissions.guard.enabled:true}")
    private boolean checkEnabled;

    @Before("@annotation(permissionAllowed)")
    public void beforePermissionValidation(PermissionAllowed permissionAllowed) {
        PermissionType[] permissions = permissionAllowed.permissionAllowed();
        Set<PermissionType> allowedPermissions = new HashSet<>(Arrays.asList(permissions));
        if (checkEnabled
                && commonUtilsService.isNotAuthorizedUser(getPrincipal(), allowedPermissions)) {
            throw new PermissionConstraintException("Not Authorized");
        }
    }
}

```